### DISCUSSION TITLE
response_status  for calendars

### DISCUSSION THREAD

**Discussion by shradhit** [O365/python-o365#995] [Q&A]
I have written a code to retrieve calendar information. I would appreciate it if you could provide guidance on how to obtain the response_status information for the invitees.

`
import datetime as dt
import pandas as pd
from O365 import Account, MSGraphProtocol
import html2text

CLIENT_ID = "xx"
CLIENT_SECRET = "yy"

credentials = (CLIENT_ID, CLIENT_SECRET)

protocol = MSGraphProtocol()
scopes = ['Calendars.Read.Shared']
account = Account(credentials, protocol=protocol)

if account.authenticate(scopes=scopes):
    print('Authenticated!')

schedule = account.schedule()
calendar = schedule.get_default_calendar()

q = calendar.new_query('start').greater_equal(dt.datetime(2023, 1, 1))
q.chain('and').on_attribute('end').less_equal(dt.datetime(2023, 12, 30))

events = calendar.get_events(limit=500, query=q, include_recurring=True)

subjects = []
starts = []
ends = []
locations = []
attendees_all = []
body = []
body_all = []
organizers = []
created_all = []
sensitivity_all = []
isAllDay_all = []
is_online_meeting_all = []
for event in events:
    subjects.append(event.subject)
    starts.append(event.start)
    ends.append(event.end)
    locations.append(event.location)

    attendee = [ (attendee.name, attendee.address, attendee.attendee_type.value)  for attendee in event.attendees ]

    #rint(attendee)
    attendees_all.append(attendee)
    #print(attendees_all)

    body = html2text.html2text(event.body)

    body_all.append(body)

    createds = event.created
    created_all.append(createds)

    sensitivity  =event.sensitivity.value
    sensitivity_all.append(sensitivity)

    isAllDay = event.is_all_day
    isAllDay_all.append(isAllDay)

   

    #optional = [attendee.email_address for attendee in event.attendees if attendee.response_status == 'TentativelyAccepted']
    #optional_attendees.append(optional)

    #none = [attendee.email_address for attendee in event.attendees if attendee.response_status == 'None']
    #none_attendees.append(none)

    is_online_meeting = event.is_online_meeting
    is_online_meeting_all.append(is_online_meeting)

    organizers.append(event.organizer)

data = {
    'Subject': subjects,
    'Start': starts,
    'End': ends,
    'Location': locations,
    'Attendees': attendees_all,
    'Body': body,
    'Organizer': organizers,
    'Created':created_all,
    'Sensitivity':sensitivity_all,
    'isAllDay':isAllDay_all,
    'is_online_meeting_all' : is_online_meeting

}

df = pd.DataFrame(data)

**Comment by alejcas** [comment]
An event attendee has a response_status property indicating this
